projects/step-frontend/src/lib/modules/timeseries/execution-page/execution-page.component.ts

```diff
@@ -120,41 +174,56 @@ export class ExecutionPageComponent implements OnInit, OnDestroy {
     if (oldInterval.value === newInterval.value) {
       return;
     }
+    // this.updatingSubscription.unsubscribe();
+    this.terminator$.next();
     if (newInterval.value) {
-      this.updatingSubscription.unsubscribe();
-      this.triggerNextUpdate(newInterval.value, of(undefined));
+      let delay = oldInterval.value === 0 ? 0 : newInterval.value;
+      this.triggerNextUpdate(delay, of(undefined));
     }
   }
 
   triggerNextUpdate(delay: number, observableToWaitFor: Observable<unknown>) {
     this.updatingSubscription = forkJoin([timer(delay), observableToWaitFor])
       .pipe(
-        tap(() => {
-          const now = new Date().getTime();
-          if (this.executionInProgress) {
-            this.performanceViewSettings!.endTime =
-              now - (this.intervalShouldBeCanceled ? 0 : this.RUNNING_EXECUTION_END_TIME_BUFFER); // if the execution is not ended, we don't fetch until the end.
-          }
-          const timeSelection = this.performanceView.getTimeRangeSelection();
-          if (timeSelection.type === RangeSelectionType.RELATIVE && timeSelection.relativeSelection) {
-            const from = now - timeSelection.relativeSelection.timeInMs;
-            timeSelection.absoluteSelection = { from: from, to: now };
-          }
-        }),
+        takeUntil(this.terminator$),
```

Comment:
It's better to put termination subjects at last position in pipe's chain.

Reply:
@dvladir is there another reason apart that it doesn't 'polute' the business flow?

Reply:
it makes sense. thank you!